core/api/src/models/ProfilePropertyRule.ts

```diff
@@ -152,21 +159,43 @@ export class ProfilePropertyRule extends LoggedModel<ProfilePropertyRule> {
   sourceGuid: string;
 
   @AllowNull(false)
-  @Default(true)
-  @Column
-  passive: boolean;
+  @Default("draft")
+  @Column(DataType.ENUM("draft", "ready"))
+  state: string;
 
   @BelongsTo(() => Source)
   source: Source;
 
   @HasMany(() => Option, "ownerGuid")
   _options: Option[]; // the underscore is needed as "options" is an internal method on sequelize instances
 
+  @BeforeValidate
+  static async ensureKey(instance: ProfilePropertyRule) {
+    if (!instance.key) {
+      instance.key = `rule ${new Date().getTime()}`;
+    }
+  }
+
   @BeforeSave
-  static async ensureSourceOptions(instance: ProfilePropertyRule) {
+  static async ensureOptions(instance: ProfilePropertyRule) {
     const source = await instance.$get("source");
     await source.validateOptions();
-    await instance.validateOptions();
+  }
+
+  @BeforeSave
+  static async updateState(instance: ProfilePropertyRule) {
+    await StateMachine.transition(instance, STATE_TRANSITIONS);
+  }
+
+  @BeforeCreate
+  static async ensureSourceIsReady(instance: ProfilePropertyRule) {
+    const source = await Source.findOne({
+      where: { guid: instance.sourceGuid },
+    });
+
+    if (source.state !== "ready") {
```

Comment:
Some of the models have `VALID_STATES` or similar. I wonder if it might be useful to have `isReady()` or something on all the models to abstract it away to other models as they evolve.

Reply:
oh! we can remove `validStates` now!